internal/webhook/v1alpha1/workspace_webhook_test.go

```diff
@@ -229,6 +229,60 @@ var _ = Describe("Workspace Webhook", func() {
 			Expect(err).NotTo(HaveOccurred())
 			Expect(warnings).To(BeEmpty())
 		})
+
+		It("should reject update that removes created-by annotation", func() {
+			oldWorkspace := workspace.DeepCopy()
+			oldWorkspace.Annotations = map[string]string{
+				controller.AnnotationCreatedBy: "original-user",
+				"other-annotation":             "other-value",
+			}
+			newWorkspace := workspace.DeepCopy()
+			newWorkspace.Annotations = map[string]string{
+				"other-annotation": "other-value",
+			}
+
+			warnings, err := validator.ValidateUpdate(ctx, oldWorkspace, newWorkspace)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("created-by annotation is immutable"))
+			Expect(warnings).To(BeEmpty())
+		})
+
+		It("should reject update that removes all annotations", func() {
+			oldWorkspace := workspace.DeepCopy()
+			oldWorkspace.Annotations = map[string]string{
+				controller.AnnotationCreatedBy: "original-user",
+				"other-annotation":             "other-value",
+			}
+			newWorkspace := workspace.DeepCopy()
+			newWorkspace.Annotations = nil
+
+			warnings, err := validator.ValidateUpdate(ctx, oldWorkspace, newWorkspace)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("created-by annotation cannot be removed"))
+			Expect(warnings).To(BeEmpty())
+		})
```

Comment:
sanity-check: do you think we need to unit test setting the `created-by` annotation to ""?

non-blocking: can be future PRs